### DISCUSSION TITLE
Using Get-PASDiscoveredAccounts

### DISCUSSION THREAD

**Discussion by krmatthe23** [pspete/psPAS#495] [Q&A]
I am attempting to write a script that will retrieve a list of Pending Accounts.  Get-PASDiscoveredAcounts does the work.  But, in spite of the published limits I keep getting the same 20K objects back from the request.  I have tried iterating through a for loop and increasing the offset with each iteration but it seems to be ignoring both the offset and limit parameters.
Our environment is:
On-Prem
PAS v12.6
Using psPAS module 6.0.4 via Powershell 7
The loop is as follows:
for ($i=0; $i -le 200000; $i = $i + 1000){
        Write-Host("The Current offset is - " + $i)
		Get-PASDiscoveredAccount -limit 1000 -offset $i -platformtype "Windows Server Local" | Export-Csv -Path $datafile -Append -Force
        Get-PASDiscoveredAccount -limit 1000 -offset $i -platformtype "Windows Domain" | Export-Csv -Path $datafile -Append -Force
	}

My guess is that I missed something very obvious in the documentation an am using the parameters incorrectly.  But, I'm not seeing it.  So more eyes are better.

**Comment by pspete** [comment]
https://cyberark.my.site.com/s/article/CPMScanner-How-to-increase-the

**Reply by krmatthe23** [reply]
Thank you for the response.  But, we already have that set to 100,000.  Still only getting about 20,000 responses from the command output.  I'm starting to lean in the direction that we just have too many pending accounts discovered.  Perhaps it is time to purge the pending accounts safe and start fresh.

**Reply by fchavez14** [reply]
I'm currently running into the same issue, MaxDisplayRecords set to 100k but unable to retrieve anything after the first 20k. @krmatthe23 were you able to resolve this issue?
